src/inspect_ai/model/_chat_message.py

```diff
@@ -131,11 +131,11 @@ def extract_reasoning(cls, data: Any) -> Any:
             # cleave apart <think> blocks
             content = data.get("content", None)
             if isinstance(content, str):
-                parsed = parse_content_with_reasoning(content)
-                if parsed:
+                content_text, reasoning = parse_content_with_reasoning(content)
+                if reasoning:
                     data["content"] = [
-                        ContentReasoning(reasoning=parsed.reasoning),
-                        ContentText(text=parsed.content),
+                        ContentReasoning(reasoning=reasoning.reasoning),
+                        ContentText(text=content_text),
```

Comment:
This, and a few changes below, are just to adapt to the signature that I made to `parse_content_with_reasoning`. I did that to make the calling code simpler and easier to chain. There is no logic change.